### ISSUE TITLE
Add static initializer for varying types.

### ISSUE THREAD

**Issue by pbrubaker** [ispc/ispc#1484]
I would like to propose adding a static initializer for varying types at the language level.  Currently this is possible by defining a uniform array and casting to a uniform pointer to varying to trigger the load.  But the syntax is clunky, and I believe adding a static initializer would be easier to use and read.

```C
export void Now()
{
    uniform float foo[] = { 7, 6, 5, 4, 3, 2, 1, 0 };
    varying float vfoo = *((varying float* uniform)&foo[0]);
}

export void Future?()
{
    varying float vfoo = { 7, 6, 5, 4, 3, 2, 1, 0 };
}
```

This would make using stdlib functions like shuffle(), that take a varying type as the permutation, much easier to use.

There would have to be some error checking obviously to make sure the initializer list doesn't exceed SIMD width.  And type checking.  And probably other things I'm not thinking of.

**Comment by dbabokin** [comment]
Looks like to make it fully usable we need also TARGET_WIDTH macro. Otherwise there's no clear way to specialize for different target width

**Comment by pbrubaker** [comment]
Can we specialize using `programCount`?

**Comment by dbabokin** [comment]
No, unfortunately it's not a preprocessor symbol.

**Comment by pbrubaker** [comment]
Ah okay, thanks for the heads up.  Looks like TARGET_WIDTH is the best option.  I also like your suggestion of defining TARGET_ELEMENT_WIDTH.

**Comment by dbabokin** [comment]
Implemented with #1503
